src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -265,9 +271,14 @@
               (malformed
                 (i18n/tru "State {0} invalid; Must specify desired state of ''signed'' or ''revoked'' for host {1}."
                   (name desired-state) subject))
-              ; this is the happy path. we have a body, it's parsable json,
-              ; and the desired_state field is one of (signed revoked)
-              [false {::json-body json-body}])
+              (let [cert-ttl (:cert_ttl json-body)]
```

Comment:
I just noticed.. here we're referencing the key `:cert_ttl` but above we add it to setting as `:ca_ttl` - was that intentional? or should this be `:ca_ttl` too?

Reply:
oh wait no, I see, we're getting it from the json body as `:cert_ttl` above

Reply:
![](https://media.giphy.com/media/Zy4m0B9oYACoU/giphy.gif)

Reply:
Yeah it's intentional, in the request body we want it to be `cert_ttl`, since we're only trying to affect the TTL of one cert, not the whole CA. Puppet doesn't currently have that granularity: `ca_ttl` controls everything (though in this case we're just overwriting it for one cert). Either there is a ticket to add a separate `cert_ttl` setting, or we should file one (I can't find it right now).